src/components/users/userProfile/EditUserProfile.js

```diff
@@ -0,0 +1,110 @@
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import React from "react";
+import { useState, useEffect } from "react";
+import client from "../../../utils/client";
+import Header from "../../Header/Header";
+import { useParams } from "react-router-dom";
+import "./style.css";
+
+const UserForm = ({ role }) => {
+  const initialFormData = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    biography: "",
+    githubUrl: "",
+  };
+
+  const [profile, setProfile] = useState(initialFormData);
+  const [editUserResponse, setEditUserResponse] = useState("");
+
+  const { id } = useParams();
+
+  useEffect(() => {
+    client
+      .get(`/user/${id}`)
+      .then((res) => {
+        setProfile(res.data.data.user);
+      })
+      .catch((err) => console.log(err.response));
+  }, [id]);
+
+  const handleChange = (event) => {
+    event.preventDefault();
+    const { value, name } = event.target;
+
+    setProfile({
+      ...profile,
+      [name]: value,
+    });
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    const putBody = {
+      firstName: profile.firstName,
+      lastName: profile.lastName,
+      email: profile.email,
+      password: profile.password,
+      bio: profile.biography,
+      githubUrl: profile.githubUrl,
+    };
+    client
+      .put("/user", putBody)
+      .then((res) => res.json)
+      .then((res) => {
+        setEditUserResponse(res.data);
```

Comment:
Does the user get any feedback when their profile has been updated? It might be worth a confirmation message, or possible redirecting back to the view profile page?

Reply:
Very good point and something I definitely over looked. Function wise, I prefer the navigation back to the profile page personally as it feels a bit neater. I have implemented that now but would you prefer a confirmation message instead?

Reply:
Navigation is good!